webapp/channels/src/components/channel_header/channel_header.tsx

```diff
@@ -517,6 +518,7 @@ class ChannelHeader extends React.PureComponent<Props, State> {
 
         let headerTextContainer;
         const headerText = (isDirect && dmUser?.is_bot) ? dmUser.bot_description : channel.header;
+        const popoverWidth = this.headerDescriptionRef.current?.clientWidth || 0 - (this.props.hasMoreThanOneTeam ? 64 : 0);
```

Comment:
You generally shouldn't access refs inside the render method of a component because:
1. Render should ideally be "pure" and only use data that's in either props or state.
2. Accessing the `clientWidth` of an element causes the browser to reflow [which can affect performance](https://developers.google.com/speed/docs/insights/browser-reflow), so we don't want to do it while React is rendering.

Instead of checking for that width on every `render`, I think we could set it once in `showChannelHeaderPopover` and then store it in the component state similar to how we store `leftOffset` and `topOffset` in there too. That way, we're keeping render pure, and we're only measuring that div once instead of on every render

Reply:
That's a good point. Thanks for a very clear explanation, I added the fix.